analyses/molecular-subtyping-chordoma/01-Subtype-chordoma.Rmd

```diff
@@ -0,0 +1,231 @@
+---
+title: "Subtyping chordoma"
+output: html_notebook
+author: Mateusz Koptyra 
+date: 20191121
+---
+
+This notebook prepares _SMARCB1_ copy number and expression data for chordoma samples for the purpose of identifying poorly-differentiated chordoma samples, which are characterized by loss of _SMARCB1_.
+
+## Set up
+
+```{r}
+library(dplyr)
+library(readr)
+library(ggplot2)
+```
+
+### Read in data
+
+```{r}
+histologies_df <- read_tsv(file.path("..", "..", "data", 
+                                     "pbta-histologies.tsv"))
+```
+```{r}
+chordoma_samples <- histologies_df %>%
+  filter(short_histology == "Chordoma") %>% 
+  pull(Kids_First_Biospecimen_ID)
+```
+
+```{r}
+# TODO: update to use consensus file and likely a more permissive version of 
+# the focal-cn-file-preparation annotation step
+# Here, we're using an older version of the annotated files that used exons
+focal_cn_df <- read_tsv("https://github.com/AlexsLemonade/OpenPBTA-analysis/raw/fa214291713575be7fd20c92374b268870f4173f/analyses/focal-cn-file-preparation/results/cnvkit_annotated_cn_autosomes.tsv.gz")
+```
+
+```{r}
+#we need to include the sample_id field from pbta-histologies.tsv in the final table (field will allow #us to map between RNA-seq (e.g., SMARCB1 expression values) and WGS data (e.g., SMARCB1 focal copy #number status) from the same event for a given individual).
+#To get the SMARCB1 jitter plot in the photo here #250 (comment), you will first need to read in the #collapsed expression data
+expression_data <- read_rds(file.path("..", "..", "data", "pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"))
+```
+
+### Output
+
+```{r}
+# scatterplot
+plot_dir <- "plots"
+if (!dir.exists(plot_dir)) {
+  dir.create(plot_dir)
+}
+plot_file <- file.path(plot_dir, "smarcb1_expression_copy_status.png")
+
+# tabular data
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+output_file <- file.path(results_dir, "chordoma_smarcb1_status.tsv")
+```
+
+## Prepare the data
+
+```{r}
+chordoma_loss <- focal_cn_df %>% 
+  filter(biospecimen_id %in% chordoma_samples, gene_symbol == "SMARCB1")
```

Comment:
Seems like great catch for me - don't know actually how to explore that.
I also sense based on the ploidy column that it may be not that straight forward for the BS_5B6XZ7YP sample.

Reply:
```r
  filter(biospecimen_id %in% chordoma_samples, gene_symbol == "SMARCB1")
```
became
```r
  filter(biospecimen_id %in% chordoma_samples, 
          gene_symbol == "SMARCB1",
          status == "loss")
```
`status == "loss"` specifies that it must be a loss and that gains or amplifications can't leak through. I can confirm that the only kind of CNA in _SMARCB1_ in chordoma samples are losses.